packages/cmf/src/sagas/component.js

```diff
@@ -33,8 +33,13 @@ export function* onSagaStart(action) {
 		);
 	} else {
 		const task = yield fork(saga, action.props);
-		yield take(`${CONST.WILL_UNMOUNT_SAGA_STOP}_${action.saga}`);
-		yield cancel(task);
+		// eslint-disable-next-line no-constant-condition
+		while (true) {
+			const stopAction = yield take(`${CONST.WILL_UNMOUNT_SAGA_STOP}_${action.saga}`);
```

Comment:
You can use the [`take(pattern)`](https://redux-saga.js.org/docs/api/index.html#takepattern) where pattern is a function matching. It can then be simplify something like :

```javascript
yield take(a =>
    a.type === `${CONST.WILL_UNMOUNT_SAGA_STOP}_${action.saga}`
    && a.event.componentId === action.event.componentId
);
yield cancel(task);
```
Without any loop and if statements